src/Components/ADT3DViewer/ADT3DViewer.tsx

```diff
@@ -623,15 +623,18 @@ const ADT3DViewerBase: React.FC<IADT3DViewerProps> = ({
         left: number,
         top: number
     ) => {
+        setAlertPanelItems({
+            element: badgeGroup.element,
+            behaviors: badgeGroup.behaviors,
+            twins: badgeGroup.twins
+        });
+        // Adding offsets to ensure the popover covers the alerts badges as per the designs
+        setAlertPopoverPosition({ left: left - 50, top: top - 30 });
         if (!isAlertPopoverVisible) {
-            setAlertPanelItems({
-                element: badgeGroup.element,
-                behaviors: badgeGroup.behaviors,
-                twins: badgeGroup.twins
-            });
-            // Adding offsets to ensure the popover covers the alerts badges as per the designs
-            setAlertPopoverPosition({ left: left - 50, top: top - 30 });
             setIsAlertPopoverVisible(true);
+        } else {
+            setIsAlertPopoverVisible(false); // Hide old popover
```

Comment:
It doesn;t really matter what order they are done in. The point of these two lines is to force a refresh.

Reply:
I was trying to avoid root causing this, but decided to do that - the set visible should be unconditional and then the issue does not arise. No need for false/true.

Reply:
The diffs are a bit hard to follow. The code is now:

    const onBadgeGroupHover = (
        badgeGroup: SceneViewBadgeGroup,
        left: number,
        top: number
    ) => {
        setAlertPanelItems({
            element: badgeGroup.element,
            behaviors: badgeGroup.behaviors,
            twins: badgeGroup.twins
        });
        // Adding offsets to ensure the popover covers the alerts badges as per the designs
        setAlertPopoverPosition({ left: left - 50, top: top - 30 });
        setIsAlertPopoverVisible(true);
    };

Reply:
The issue was caused if the adapter polled whilst the popover was visible - some stale state going on